### PR TITLE
Add link to GitHub Apps authentication package in docs.

### PR DESCRIPTION
**Pull request by bradleyfalzon**
As discussed in #627 I have a package which provides the GitHub Apps Authentication as an Installation `http.RoundTripper` that I use with this GitHub library successfully. I'm happy to keep maintaining it in an external repository, but a link from this library may assist others in finding the package.

Relates to #455.

### CODE REVIEW

**Comment by shurcooL** [comment]
I still feel it'd be nice to replace the period with a colon and add a simple basic usage example here, in this README. Just like there's a short example for `oauth2` above.

You said:

> I've sent through a minor one-liner, as I did begin to write a larger section which mostly just duplicated the docs already in https://github.com/bradleyfalzon/ghinstallation, so just pared it back to a minor note. Let me know in the PR if you'd like me to expand upon the docs though, happy to.

The way I see it, in this README, just include a basic usage example for people to get started. But leave advanced usage, details, explanations in the ghinstallation repo README.

Just like for oauth2, this repo README constrains a basic snippet for getting started, but oauth2 package docs contain more thorough detailed docs.

**Comment by bradleyfalzon** [comment]
Sure, I can add a simple example.

Is this OK:

```
// Shared transport to reuse TCP connections.
tr := &http.Transport{}

// Wrap the shared transport for use with the integration ID 1 authenticating with installation ID 99.
itr, err := ghinstallation.NewKeyFromFile(tr, 1, 99, "2016-10-19.private-key.pem")

// Use installation transport with new client.
client := github.NewClient(http.Client{Transport: itr})
```

I can make it inline the `*http.Transport` if you like? The oauth2 example is a complete working program, would you prefer that too?

**Comment by shurcooL** [comment]
I think it'd be helpful to include the import statement. It makes it easier to see the whole thing at a glance. Otherwise it's extra work and uncertainty for people to figure out the import path (it could be a vanity import path, so it requires extra steps to find out).

You can stop after creating a GitHub client, no need to call methods on it. Maybe just a `// Use client...` comment.

Also, why do you suggest creating a separate `http.RoundTripper` for it, instead of just using [`http.DefaultTransport`](https://godoc.org/net/http#DefaultTransport)? The zero value of `http.Transport{}` doesn't seem to have good defaults.

**Comment by bradleyfalzon** [comment]
Sure, updated.

> Also, why do you suggest creating a separate http.RoundTripper for it, instead of just using http.DefaultTransport? The zero value of http.Transport{} doesn't seem to have good defaults.

Ah must just be an oversight, GopherCI uses DefaultTransport (https://github.com/bradleyfalzon/gopherci/blob/03badd2364a4a3e85c0e4389c88ca86b7e2f7a70/internal/github/github.go#L38), so I'll modify that example.

File: README.md [hunk 1]
```diff
@@ -74,6 +74,26 @@ See the [oauth2 docs][] for complete instructions on using that library.
...
 [`BasicAuthTransport`](https://godoc.org/github.com/google/go-github/github#BasicAuthTransport).
 
+GitHub Apps authentication can be provided by [ghinstallation](https://github.com/bradleyfalzon/ghinstallation).
+
+```go
+import "github.com/bradleyfalzon/ghinstallation"
+
+func main() {
+  // Shared transport to reuse TCP connections.
+  tr := http.DefaultTransport
```
Comment (shurcooL) [inline README.md]: I think you can delete this line and its comment, and pass `http.DefaultTransport` directly to the `ghinstallation.NewKeyFromFile` call below.

It's pretty common knowledge that transports should be reused, unless there are specific reasons not to. I don't think you need to elaborate on that here. This is meant to be a minimal simple example to get started quickly.

File: README.md [hunk 2]
```diff
@@ -74,6 +74,26 @@ See the [oauth2 docs][] for complete instructions on using that library.
...
+
+  // Wrap the shared transport for use with the integration ID 1 authenticating with installation ID 99.
+  itr, err := ghinstallation.NewKeyFromFile(tr, 1, 99, "2016-10-19.private-key.pem")
+  if err != nil {
+    log.Fatal(err)
```
Comment (shurcooL) [inline README.md]: Replace with `// handle error` to leave error handling up to user. We don't want to suggest bad practices, and doing better error handling can be verbose. It's common for usage examples to not elaborate on error handling, just that it should be done.

File: README.md [hunk 3]
```diff
@@ -74,6 +74,26 @@ See the [oauth2 docs][] for complete instructions on using that library.
...
+  }
+
+  // Use installation transport with client
+  client := github.NewClient(http.Client{Transport: itr})
+}
```
Comment (shurcooL) [inline README.md]: Replace `}` with `// use client...`. This is an incomplete program anyway (it's missing a `package` clause), so leaving out the closing `}` is okay. I don't see why you'd want to include `}` and make it seem like `client` isn't used at all.

Comment (bradleyfalzon) [inline README.md]: Yeah, but I was half copying the style of oauth2, I'll make this change, but let me know if you want to make further edits there for consistency.

File: README.md (hunk 2 above)
Comment (bradleyfalzon) [inline README.md]: Sure, done.

File: README.md (hunk 1 above)
Comment (bradleyfalzon) [inline README.md]: Agreed, done.

File: README.md [hunk 4]
```diff
@@ -74,6 +74,22 @@ See the [oauth2 docs][] for complete instructions on using that library.
...
 
+GitHub Apps authentication can be provided by [ghinstallation](https://github.com/bradleyfalzon/ghinstallation).
+
+```go
+import "github.com/bradleyfalzon/ghinstallation"
+
+func main() {
+  // Wrap the shared transport for use with the integration ID 1 authenticating with installation ID 99.
+  itr, err := ghinstallation.NewKeyFromFile(http.DefaultTransport, 1, 99, "2016-10-19.private-key.pem")
+  // Handle error.
```
Comment (shurcooL) [inline README.md]: I meant like this:

```Go
itr, err := ghinstallation.NewKeyFromFile(http.DefaultTransport, 1, 99, "2016-10-19.private-key.pem")
if err != nil {
	// Handle error.
}
```

I think it's a lot more readable at a glance this way, compared to just a plain text comment without the `if err != nil {` line.

File: github/doc.go [hunk 5]
```diff
@@ -63,6 +63,21 @@ See the oauth2 docs for complete instructions on using that library.
...
+	func main() {
+		// Wrap the shared transport for use with the integration ID 1 authenticating with installation ID 99.
+		itr, err := ghinstallation.NewKeyFromFile(http.DefaultTransport, 1, 99, "2016-10-19.private-key.pem")
+		// Handle error.
+
+		// Use installation transport with client
+		client := github.NewClient(http.Client{Transport: itr})
+
+		// Use client...
+
```
Comment (shurcooL) [inline github/doc.go]: Okay, I see how this looks kinda bad/unbalanced without the closing `}` for `func main() {` now. So let's add that back in. Keep the `// Use client...` comment of course.

File: README.md [hunk 6]
```diff
@@ -74,6 +74,22 @@ See the [oauth2 docs][] for complete instructions on using that library.
 For API methods that require HTTP Basic Authentication, use the
 [`BasicAuthTransport`](https://godoc.org/github.com/google/go-github/github#BasicAuthTransport).
 
+GitHub Apps authentication can be provided by [ghinstallation](https://github.com/bradleyfalzon/ghinstallation).
```
Comment (shurcooL) [inline README.md]: Add the word "package" after ghinstallation maybe?

> GitHub Apps authentication can be provided by the [ghinstallation](https://github.com/bradleyfalzon/ghinstallation) package.

Notice how above it says:

> The easiest and recommended way to do this is using the [oauth2](https://github.com/golang/oauth2) **library**, ...

Comment (bradleyfalzon) [inline README.md]: Agreed, done.

File: README.md (hunk 4 above)
Comment (bradleyfalzon) [inline README.md]: Of course, I wasn't thinking. Done.

File: github/doc.go (hunk 5 above)
Comment (bradleyfalzon) [inline github/doc.go]: Yeah sounds good. Done.

**Comment by bradleyfalzon** [comment]
> are we okay with adding endorsement for the ghinstallation package in a separate repository?

The other option is to roll ghinstallation into go-github, I'd have no problem with that. Potentially with the idea to split the single Transport package into IntegrationTransport and InstallationTransport.

File: README.md [hunk 7]
```diff
@@ -74,6 +74,26 @@ See the [oauth2 docs][] for complete instructions on using that library.
...
+
+func main() {
+  // Wrap the shared transport for use with the integration ID 1 authenticating with installation ID 99.
+  itr, err := ghinstallation.NewKeyFromFile(http.DefaultTransport, 1, 99, "2016-10-19.private-key.pem")
+  if err != nil {
+    // Handle error.
+  }
+
+  // Use installation transport with client.
+  client := github.NewClient(http.Client{Transport: itr})
```
Comment (shurcooL) [inline README.md]: Minor. Shouldn't this be `&http.Client`?

Comment (bradleyfalzon) [inline README.md]: Yes it should, fixed, and thanks.

**Comment by shurcooL** [comment]
Friendly ping @gmlewis and @willnorris. I want to make sure you saw [my comment from 7 days ago](https://github.com/google/go-github/pull/649#pullrequestreview-40645150), in case it got buried by the minor typo discussion that followed.

**Comment by gmlewis** [comment]
I'm fine with the endorsement, but before proceeding, I would like confirmation from @willnorris and I defer to his judgement. I'll ask him to review this.

**Comment by shurcooL** [comment]
Great! Merging.

If there are any important updates or changes to the linked package, please let us know @bradleyfalzon. We're here to help. Thanks for creating it!

**Pull request merged**